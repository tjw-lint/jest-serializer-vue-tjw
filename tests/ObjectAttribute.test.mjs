const helpers = require('./helpers.js');

import { shallowMount } from '@vue/test-utils';
import ObjectAttribute from './components/ObjectAttribute.vue';

describe('ObjectAttribute.vue', () => {
  test('Does not mutate original component wrapper', () => {
    helpers.mockSettings({ stringifyObjects: true });

    const wrapper = shallowMount(ObjectAttribute);

    expect(wrapper.html())
      .toMatchSnapshot('1 wrapper.html()');

    expect(wrapper)
      .toMatchSnapshot('2 wrapper');

    expect(wrapper.html())
      .toMatchSnapshot('3 wrapper.html()');

    expect(wrapper)
      .toMatchSnapshot('4 wrapper');

    expect(wrapper.find('[data-test="div"]').element.title)
      .toEqual('[object Object]');

    expect(wrapper.find('[data-test="h1"]').element.title)
      .toEqual('asdf,qwer');
  });

  test('Date mode', () => {
    helpers.mockSettings({ stringifyObjects: true });

    const wrapper = shallowMount(ObjectAttribute, {
      propsData: {
        dateMode: true
      }
    });

    expect(wrapper)
      .toMatchSnapshot();
  });
});
