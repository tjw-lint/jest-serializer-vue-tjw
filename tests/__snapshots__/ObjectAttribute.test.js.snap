// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectAttribute.vue Date mode 1`] = `
<div>
  <hr title="1577836800000">
</div>
`;

exports[`ObjectAttribute.vue Does not mutate original component wrapper: 1 wrapper.html() 1`] = `
<div title="[object Object]">
  <h1 title="asdf,qwer" class="active">
    text
  </h1>
  <hr title="text,22">
  <hr>
  <hr title="true">
  <hr title="Invalid Date">
  <hr title="Error: err">
  <hr title="Infinity">
  <hr title="-Infinity">
  <hr title="NaN">
  <hr>
  <hr title="75">
  <hr title="[object Object]">
  <hr title="[object Set]">
  <hr title="Text">
  <hr title="function String() { [native code] }">
  <hr>
  <hr title="Text
Text">
</div>
`;

exports[`ObjectAttribute.vue Does not mutate original component wrapper: 2 wrapper 1`] = `
<div title="{a:true,b:'Some \\'text\\'',c:{d:{e:'23'},f:'\\'&quot;\\'&quot;'}}">
  <h1 title="['asdf','qwer']" class="active">
    text
  </h1>
  <hr title="['text',22]">
  <hr title="false">
  <hr title="true">
  <hr title="Invalid Date">
  <hr title="Error: err">
  <hr title="Infinity">
  <hr title="-Infinity">
  <hr title="NaN">
  <hr title="null">
  <hr title="75">
  <hr title="{a:{c:true},b:{d:'text'}}">
  <hr title="['a','b','c']">
  <hr title="Text">
  <hr title="Function">
  <hr title="undefined">
  <hr title="Text
Text">
</div>
`;

exports[`ObjectAttribute.vue Does not mutate original component wrapper: 3 wrapper.html() 1`] = `
<div title="[object Object]">
  <h1 title="asdf,qwer" class="active">
    text
  </h1>
  <hr title="text,22">
  <hr>
  <hr title="true">
  <hr title="Invalid Date">
  <hr title="Error: err">
  <hr title="Infinity">
  <hr title="-Infinity">
  <hr title="NaN">
  <hr>
  <hr title="75">
  <hr title="[object Object]">
  <hr title="[object Set]">
  <hr title="Text">
  <hr title="function String() { [native code] }">
  <hr>
  <hr title="Text
Text">
</div>
`;

exports[`ObjectAttribute.vue Does not mutate original component wrapper: 4 wrapper 1`] = `
<div title="{a:true,b:'Some \\'text\\'',c:{d:{e:'23'},f:'\\'&quot;\\'&quot;'}}">
  <h1 title="['asdf','qwer']" class="active">
    text
  </h1>
  <hr title="['text',22]">
  <hr title="false">
  <hr title="true">
  <hr title="Invalid Date">
  <hr title="Error: err">
  <hr title="Infinity">
  <hr title="-Infinity">
  <hr title="NaN">
  <hr title="null">
  <hr title="75">
  <hr title="{a:{c:true},b:{d:'text'}}">
  <hr title="['a','b','c']">
  <hr title="Text">
  <hr title="Function">
  <hr title="undefined">
  <hr title="Text
Text">
</div>
`;
